src/concurrency/vector_clock.rs

```diff
@@ -448,7 +467,13 @@ mod tests {
         while let Some(0) = slice.last() {
             slice = &slice[..slice.len() - 1]
         }
-        VClock(slice.iter().copied().map(|time| VTimestamp { time, span: DUMMY_SP }).collect())
+        VClock(
+            slice
+                .iter()
+                .copied()
+                .map(|time| VTimestamp { time_and_read_type: time.shl(1), span: DUMMY_SP })
```

Comment:
Please don't duplicate the encoding logic here, add a constructor instead or use `set_time`.